scripts/simple-scylla-cluster-manager.py

```diff
@@ -0,0 +1,776 @@
+        self._logger.debug(f"Loaded config: {json.dumps(self._config, indent=4)}")
+        return self._make_success()
+
+    def _get_node_config_item(self, i, key):
+        node_config = self._config["nodes"][i]
+        if key in node_config:
+            return node_config[key]
+
+        return self._config["default_scylla_config"][key]
+
+    def _send_reply(self, res = None):
+        if res is None:
+            res = self._make_success()
+        self._out.write(json.dumps(res) + '\n')
+        self._out.flush()
+
+    def start_cluster(self, **kwargs):
+        self._logger.info(f"start_cluster({kwargs})")
+        pass
+
+    def add_node(self, **kwargs):
+        self._logger.info(f"add_node({kwargs})")
+
+        i = self._get_next_index()
+        node_dir = f"node{i}"
+        os.mkdir(node_dir)
+
+        os.mkdir(os.path.join(node_dir, "data"))
+        os.mkdir(os.path.join(node_dir, "commitlog"))
+        os.mkdir(os.path.join(node_dir, "hints"))
+        os.mkdir(os.path.join(node_dir, "view_hints"))
+
+        node_ip = f"127.{self._config['network_prefix']}.0.{i}"
+
+        default_node_config = self._config["default_scylla_config"]
+        node_config = {
+            "index": i,
+            "ip": node_ip,
+        }
+        for item_name in default_node_config.keys():
+            value = kwargs.get(item_name, None)
+            if value is None or (type(value) == list and not value):
+                continue
+            node_config[item_name] = value
+
+        self._config["nodes"][i] = node_config
+
+        self._logger.debug(f"node_config is {node_config}")
+
+        scylla_path = self._get_node_config_item(i, "path")
+        scylla_mode = self._get_node_config_item(i, "mode")
+        shutil.copy(os.path.join(scylla_path, "build", scylla_mode, "scylla"), os.path.join(node_dir, "scylla"))
+        shutil.copy(os.path.join(scylla_path, "conf", "scylla.yaml"), os.path.join(node_dir, "scylla.yaml"))
+        shutil.copytree(os.path.join(scylla_path, "api", "api-doc"), os.path.join(node_dir, "api-doc"))
+
+        self._save_config()
+
+        return self._make_success(f"Added node with index {i}")
+
+    def _maybe_apply_all(self, callback, **kwargs):
+        self._logger.debug(f"_maybe_apply_all({kwargs})")
+        if "all" in kwargs and kwargs["all"]:
+            if "node_index" in kwargs and not kwargs["node_index"] is None:
+                raise RuntimeError("Cannot pass both --node-index and --all to node operation")
+            kwargs["all"] = False
+            for i in self._config["nodes"].keys():
+                kwargs["node_index"] = i
+                callback(**kwargs)
+            return True
+        elif not "node_index" in kwargs or not kwargs["node_index"] is None:
+            return False
+        else:
+            raise RuntimeError("Either --node-index or --all has to be passed to node operation")
+
+    def start_node(self, **kwargs):
```

Comment:
why not naming the arguments explicitly? at least we can name the ones consumed by this function and pass the others down to the subroutines called by it for better readability.

Reply:
Done in v2.